### PR TITLE
Update pod spec to ensure guaranteed CPU

### PR DESCRIPTION
**Pull request by imiller0**
Pod spec updated for guaranteed CPU scheduling and additional parameters
to improve performance.

Signed-off-by: Ian Miller <imiller@redhat.com>

### CODE REVIEW

File: pod.j2 [hunk 1]
```diff
@@ -19,17 +19,24 @@ spec:
...
         - --mlockall
-        - --priority=80
-        - --interval=200
+        - --priority=95
+        - --interval=1000
         - --distance=0
+        - --smi
+        - -a
+        - -t 16
+        - -n
```
Comment (Avielyo10) [inline pod.j2:37]: `--smp`: Standard SMP testing: options -a -t -n and same priority of all threads
`-t [NUM] --threads=NUM`: number of threads: without NUM, threads = max_cpus
Therefore `--smp` == `-a -t 16 -n`

File: pod.j2 [hunk 2]
```diff
@@ -19,17 +19,24 @@ spec:
...
+        - -h 30
```
Comment (Avielyo10) [inline pod.j2]: `-h       --histogram=US`: dump a latency histogram to stdout after the run US is the max latency time to be be tracked in microseconds. This option runs all threads at the same priority.
`-H       --histofall=US`: same as `-h` except with an additional summary column
Therefore `--histofall=US`, which we can set to `--histofall=30` instead.

**Review by Avielyo10 — CHANGES_REQUESTED** [review]

File: pod.j2 [hunk 3]
```diff
@@ -19,17 +19,24 @@ spec:
...
+        - --refresh_on_max
```
Comment (Avielyo10) [inline pod.j2:38]: This won't generate the histogram dump, are you sure?

**Review by Avielyo10 — CHANGES_REQUESTED** [review]

File: pod.j2 (hunk 3 above)
Comment (Avielyo10) [reply pod.j2:38]: Well after I tested it with 2G memory & priority=95 & interval=1000 we get the dump

File: pod.j2 [hunk 4]
```diff
@@ -19,17 +19,24 @@ spec:
...
       command: ["cyclictest"]
       args:
-        - --smp
         - --duration={{ duration }}
         - --histofall=US
         - --mlockall
-        - --priority=80
-        - --interval=200
+        - --priority=95
+        - --interval=1000
```
Comment (Avielyo10) [inline pod.j2:32]: Those gave very bad results, `Max` gets 70~ on avg
Are you sure?

File: pod.j2 [hunk 5]
```diff
@@ -19,17 +19,24 @@ spec:
...
         - --distance=0
+        - --smi
```
Comment (Avielyo10) [inline pod.j2:34]: Shows consistent `0` under it's column

File: pod.j2 (hunk 1 above)
Comment (imiller0) [reply pod.j2:37]: --smp recognizes the CPU topology of the system and not of the pod. So --smp ends up running 80 threads in the 16 CPU pod. I broke it out as -a -t 16 -n to work around that.

File: pod.j2 (hunk 2 above)
Comment (imiller0) [reply pod.j2]: We can leave just the --histofall=US

File: pod.j2 (hunk 1 above)
Comment (Avielyo10) [reply pod.j2:37]: OK

File: pod.j2 (hunk 2 above)
Comment (Avielyo10) [reply pod.j2]: --histofall=US was my bug, see https://github.com/Avielyo10/cyclictest/commit/10fccb2689dced6a95d51e4d92ad1a61b63d5a57

**Pull request merged**